crates/ide-assists/src/handlers/generate_impl.rs

```diff
@@ -1,10 +1,27 @@
 use syntax::{
-    ast::{self, make, AstNode, HasName},
+    ast::{self, edit_in_place::Indent, make, AstNode, HasName},
     ted,
 };
 
 use crate::{utils, AssistContext, AssistId, AssistKind, Assists};
 
+fn insert_impl(impl_: ast::Impl, nominal: &ast::Adt) {
+    // Add a blank line after the adt
+    let mut elems = Vec::with_capacity(3);
+    elems.push(make::tokens::blank_line().into());
+
+    // Add the indentation of the impl
+    let indent = nominal.indent_level();
+    if !indent.is_zero() {
+        let whitespace = make::tokens::whitespace(&indent.to_string());
+        elems.push(whitespace.into());
+    }
```

Comment:
I believe doing something like
```rust
make::tokens::whitespace(&format!("\n\n{indent}")).into()
```
can do the same thing (see https://github.com/rust-lang/rust-analyzer/blob/d5a424f3389007b4881b91e070fb4d9b3d722c0c/crates/ide-assists/src/handlers/extract_type_alias.rs#L80). Also means that once we have an indenter to fixup.indemtation for code assists, it'd be quite simple to remove the manual indentation fixup.

Reply:
That seems a lot simpler as well, I've pushed a fix to use this method. Thanks for the feedback!